app/src/main/java/com/example/shiv/mdbpokedexproject2/FilterActivity.java

```diff
@@ -0,0 +1,387 @@
+package com.example.shiv.mdbpokedexproject2;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CompoundButton;
+import android.widget.EditText;
+import android.widget.Toast;
+import android.widget.ToggleButton;
+
+import java.util.ArrayList;
+
+public class FilterActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_filter);
+
+        //Setup buttons and fields
+        final EditText min_Attack = (EditText) findViewById(R.id.editText2);
+        final EditText min_Defense = (EditText) findViewById(R.id.editText4);
+        final EditText min_health = (EditText) findViewById(R.id.editText5);
+        Button button2 = (Button) findViewById(R.id.button2);
+        ToggleButton dragon = (ToggleButton) findViewById(R.id.toggleButtonDragon);
+        ToggleButton fairy = (ToggleButton) findViewById(R.id.toggleButtonFairy);
+        ToggleButton normal = (ToggleButton) findViewById(R.id.toggleButtonNormal);
+        ToggleButton fire = (ToggleButton) findViewById(R.id.toggleButtonFire);
+        ToggleButton water = (ToggleButton) findViewById(R.id.toggleButtonWater);
+        ToggleButton grass = (ToggleButton) findViewById(R.id.toggleButtonGrass);
+        ToggleButton electric = (ToggleButton) findViewById(R.id.toggleButtonElectric);
+        ToggleButton ice = (ToggleButton) findViewById(R.id.toggleButtonIce);
+        ToggleButton ground = (ToggleButton) findViewById(R.id.toggleButtonGround);
+        ToggleButton flying = (ToggleButton) findViewById(R.id.toggleButtonFlying);
+        ToggleButton poison = (ToggleButton) findViewById(R.id.toggleButtonPoison);
+        ToggleButton fighting = (ToggleButton) findViewById(R.id.toggleButtonFighting);
+        ToggleButton psychic = (ToggleButton) findViewById(R.id.toggleButtonPsychic);
+        ToggleButton dark = (ToggleButton) findViewById(R.id.toggleButtonDark);
+        ToggleButton rock = (ToggleButton) findViewById(R.id.toggleButtonRock);
+        ToggleButton bug = (ToggleButton) findViewById(R.id.toggleButtonBug);
+        ToggleButton ghost = (ToggleButton) findViewById(R.id.toggleButtonGhost);
+        ToggleButton steel = (ToggleButton) findViewById(R.id.toggleButtonSteel);
+        ToggleButton allTypes = (ToggleButton) findViewById(R.id.toggleButtonAllTypes);
+
+
+
+
+
+
+
+        min_Attack.setText("" + MainActivity.min_attack);
+        min_Defense.setText("" + MainActivity.min_defense);
+        min_health.setText("" + MainActivity.min_health);
+
+
+
+        button2.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
```

Comment:
Yeah, so keep the catch block, just don't put anything in it. It doesn't change the logic, just the readability. Should probably also change Throwable to be more specific, probably NullPointerException